safe_transaction_service/events/services/queue_service.py

```diff
@@ -37,34 +37,29 @@ def connect(self) -> Optional[BlockingConnection]:
                 exchange_type=ExchangeType.fanout,
                 durable=True,
             )
-            # Send messages if there was any missing
-            # self.send_unsent_events()
             logger.debug("Opened connection to RabbitMQ")
             return self.connection
-        except pika.exceptions.AMQPConnectionError:
+        except pika.exceptions.AMQPError:
             logger.error("Cannot open connection to RabbitMQ")
             return None
 
-    def is_connected(self) -> bool:
-        """
-        :return: `True` if connected, `False` otherwise
-        """
-        return self.connection and self.connection.is_open
-
-    def publish(self, message: str) -> bool:
+    def publish(self, message: str, retry: Optional[bool] = True) -> bool:
         """
         :param message:
+        :param retry:
         :return: `True` if message was published, `False` otherwise
         """
-        # Check if is still connected if not try to reconnect
-        if not self.is_connected() and not self.connect():
-            return False
         try:
             self.channel.basic_publish(
                 exchange=self.exchange_name, routing_key="", body=message
             )
             return True
-        except pika.exceptions.AMQPConnectionError:
+        except pika.exceptions.AMQPError:
+            if retry:
+                logger.info("The connection has been terminated. trying again.")
+                # One more chance
+                self.connect()
+                self.publish(message, retry=False)
```

Comment:
You are returning `False` even if you manage to reconnect!

```suggestion
                return self.publish(message, retry=False)
```